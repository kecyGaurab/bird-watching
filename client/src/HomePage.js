/* eslint-disable react/jsx-closing-tag-location */
/* eslint-disable react/jsx-indent */
/* eslint-disable react/jsx-no-undef */
import React from 'react';
import { Link, withRouter } from 'react-router-dom';

import { Container, Grid, Snackbar, Button } from '@material-ui/core';
import MuiAlert from '@material-ui/lab/Alert';
import Bird from './components/Bird/bird';

const HomePage = ({ birds, ...props }) => {
  const Alert = () => {
    return <MuiAlert elevation={6} variant="filled" {...props} />;
  };

  const { handleClose, open, message, handleRemove, error, sortedBirds, filteredBirds } = {
    ...props,
  };
  return (
    <Container>
      <Grid justify="space-around" container direction="row" spacing={3}>
        <Grid item xs={12}>
          <Snackbar
            anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
            open={open}
            autoHideDuration={4000}
            onClose={handleClose}
          >
            {error ? (
              <Alert onClose={handleClose} severity="error">
                {message}
              </Alert>
            ) : (
              <Alert onClose={handleClose} severity="success">
                {message}
              </Alert>
            )}
          </Snackbar>
        </Grid>
        <Grid item xs={11}>
          <Link style={{ textDecoration: 'none' }} to="/add">
            <Button variant="outlined">Add New</Button>
          </Link>
        </Grid>
        {filteredBirds
          ? filteredBirds.map((b) => (
              <Grid key={b.id} item xs={12} md={3}>
                <Bird handleRemove={handleRemove} bird={b} />
              </Grid>
            ))
          : sortedBirds &&
            sortedBirds.map((b) => (
              <Grid key={b.id} item xs={12} md={3}>
                <Bird handleRemove={handleRemove} bird={b} />
              </Grid>
            ))}
      </Grid>
    </Container>
  );
};

export default withRouter(HomePage);
