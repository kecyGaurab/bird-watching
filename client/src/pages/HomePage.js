/* eslint-disable react/jsx-closing-tag-location */
/* eslint-disable react/jsx-indent */
/* eslint-disable react/jsx-no-undef */
import React, { useEffect, useState } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { Container, Grid, Button } from '@material-ui/core';
import { useDispatch, connect } from 'react-redux';
import Bird from '../components/Bird/bird';
import { initializeBirds } from '../redux/reducers/birdReducer';
import NavBar from '../components/navBar';

const mapStateToProps = (state) => {
  return {
    birds: state.observations.birds,
  };
};

const HomePage = (props) => {
  const [query, setQuery] = useState('');
  const [filteredBirds, setFilteredBirds] = useState('');
  const { birds } = props;
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(initializeBirds());
  }, [dispatch]);

  useEffect(() => {
    const handleFilter = () => {
      const matchedBirds = birds.filter((n) =>
        n.commonname.toLowerCase().startsWith(query.toLowerCase()),
      );
      setFilteredBirds(matchedBirds);
    };
    handleFilter();
  }, [query, birds]);

  const handleQueryChange = (event) => {
    setQuery(event.target.value);
  };

  const sortedBirds = birds.sort((a, b) => new Date(b.date) - new Date(a.date));

  return (
    <>
      <NavBar query={query} handleQueryChange={handleQueryChange} />
      <Container>
        <Grid justify="space-around" container direction="row" spacing={5}>
          <Grid item xs={12} align="center">
            <Link style={{ textDecoration: 'none' }} to="/add">
              <Button size="large" color="primary" variant="outlined">
                Add New
              </Button>
            </Link>
          </Grid>
          {filteredBirds
            ? filteredBirds.map((b) => (
                <Grid key={b.id} item xs={12} md={3}>
                  <Bird bird={b} />
                </Grid>
              ))
            : sortedBirds &&
              sortedBirds.map((b) => (
                <Grid key={b.id} item xs={12} md={3}>
                  <Bird bird={b} />
                </Grid>
              ))}
        </Grid>
      </Container>
    </>
  );
};

export default connect(mapStateToProps)(withRouter(HomePage));
