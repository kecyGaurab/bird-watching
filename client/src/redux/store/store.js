/* eslint-disable import/no-extraneous-dependencies */
import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

import birdReducer from '../reducers/birdReducer';

const reducer = combineReducers({
  bird: birdReducer,
});

const store = createStore(reducer, composeWithDevTools(applyMiddleware(thunk)));

export default store;
