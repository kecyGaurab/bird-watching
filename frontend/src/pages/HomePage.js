import React, { useEffect, useState } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { Container, Grid, Button } from '@material-ui/core';
import { useDispatch, useSelector } from 'react-redux';
import Bird from '../components/bird/Bird';
import { initializeBirds } from '../redux/reducers/birdReducer';
import NavBar from '../components/NavBar';
import { setCurrentUser } from '../redux/reducers/userReducer';
import Hero from '../components/common/Hero';
import Footer from '../components/common/Footer';

const HomePage = () => {
  const dispatch = useDispatch();

  const birds = useSelector((state) => state.observations.birds);
  const user = useSelector((state) => state.user);

  const [query, setQuery] = useState('');
  const [filteredBirds, setFilteredBirds] = useState('');

  useEffect(() => {
    dispatch(initializeBirds());

    const loggedUserJSON = window.localStorage.getItem('loggedUser');
    if (loggedUserJSON) {
      const savedUser = JSON.parse(loggedUserJSON);
      dispatch(setCurrentUser(savedUser));
    }

    const handleFilter = () => {
      const matchedBirds = birds.filter((n) =>
        n.commonname.toLowerCase().includes(query.toLowerCase()),
      );
      setFilteredBirds(matchedBirds);
    };
    handleFilter();
  }, [dispatch, query, birds]);

  const handleQueryChange = (event) => {
    setQuery(event.target.value);
  };

  const sortedBirds = [...birds.sort((a, b) => new Date(b.date) - new Date(a.date))];

  return (
    <>
      <NavBar query={query} handleQueryChange={handleQueryChange} />
      <Container>
        <Grid justify="space-around" container direction="row" spacing={2}>
          <Grid item xs={12} align="center">
            <Hero>
              <Link style={{ textDecoration: 'none' }} to={user.currentUser ? '/add' : '/login'}>
                <Button size="large" color="primary" variant="outlined">
                  Add New Observation
                </Button>
              </Link>
            </Hero>
          </Grid>
          {filteredBirds
            ? filteredBirds.map((bird) => (
                <Grid key={bird.id} item xs={12} md={3}>
                  <Bird bird={bird} />
                </Grid>
              ))
            : sortedBirds &&
              sortedBirds.map((b) => (
                <Grid key={b.id} item xs={12} md={3}>
                  <Bird bird={b} />
                </Grid>
              ))}
        </Grid>
      </Container>
      <Footer />
    </>
  );
};

export default withRouter(HomePage);
