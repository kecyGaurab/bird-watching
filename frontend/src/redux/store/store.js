/* eslint-disable import/no-extraneous-dependencies */
import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

import { birdReducer } from '../reducers/birdReducer';
import userReducer from '../reducers/userReducer';

const reducer = combineReducers({
  observations: birdReducer,
  user: userReducer,
});
const store = createStore(reducer, composeWithDevTools(applyMiddleware(thunk)));

export default store;
 