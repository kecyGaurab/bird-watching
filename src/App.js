import React, { Fragment, useState, useEffect } from "react";

import { Grid, Container, CssBaseline } from "@material-ui/core";
import NavBar from "./components/navBar";
import Form from "./components/form";

const App = () => {
  const [bird, setBird] = useState({ name: "", species: "" });

  const handleChange = e => {
    setBird({ ...bird, [e.target.name]: e.target.value });
  };

  const handleSubmit = e => {
    e.preventDefault();
    const data = new FormData(bird);
  };

  return (
    <Fragment>
      <CssBaseline />
      <NavBar />
      <Container>
        <Form
          handleSubmit={handleSubmit}
          handleChange={handleChange}
          bird={bird}
        />
      </Container>
    </Fragment>
  );
};

export default App;
