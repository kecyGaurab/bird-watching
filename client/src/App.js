import React from 'react';
import { Switch, Route, withRouter } from 'react-router-dom';
import Form from './components/Form/form';
import HomePage from './pages/HomePage';
import BirdDetail from './pages/BirdDetail';
import EditBird from './pages/EditBird';

const App = () => {
  return (
    <>
      <Route exact path="/" component={HomePage} />
      <Switch>
        <Route path="/add" component={Form} />
        <Route path="/:id" component={BirdDetail} />
      </Switch>
      <Route path="/:id/edit" component={EditBird} />
    </>
  );
};

export default withRouter(App);
