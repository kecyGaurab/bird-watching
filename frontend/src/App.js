/* eslint-disable no-shadow */
import React from 'react';
import { Switch, Route, withRouter } from 'react-router-dom';
import AddBird from './components/form/AddBird';
import HomePage from './pages/HomePage';
import BirdDetail from './pages/BirdDetail';
import EditBird from './pages/EditBird';
import Login from './components/form/Login';
import SignUp from './components/form/SignUp';

const App = () => {
  return (
    <>
      <Route exact path={['/', '/add']} component={HomePage} />
      <Switch>
        <Route exact path="/signup" component={SignUp} />
        <Route exact path="/login" component={Login} />
        <Route exact path="/add" component={AddBird} />
        <Route path="/:id" component={BirdDetail} />
      </Switch>
      <Route path="/:id/edit" component={EditBird} />
    </>
  );
};

export default withRouter(App);
