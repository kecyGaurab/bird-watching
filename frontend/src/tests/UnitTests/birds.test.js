import React from 'react';
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { createMemoryHistory } from 'history';
import { configure, shallow } from 'enzyme';
import { render } from '@testing-library/react';
import Adapter from 'enzyme-adapter-react-16';
import store from '../../redux/store/store';
import Login from '../../components/form/login';
import App from '../../App';
import HomePage from '../../pages/HomePage';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  shallow(<App />);
});

describe('<HomePage />', () => {
  it('Renders <HomePage /> component correctly', () => {
    const history = createMemoryHistory();
    const component = render(
      <Router history={history} location="/">
        <Provider store={store}>
          <HomePage />
        </Provider>
      </Router>,
    );
    expect(component.container).toHaveTextContent('Add New');
  });
});

describe('<Login />', () => {
  const history = createMemoryHistory();
  it('Renders <Login /> component correctly', () => {
    const tree = renderer
      .create(
        <Router history={history} location="/login">
          <Provider store={store}>
            <Login />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
