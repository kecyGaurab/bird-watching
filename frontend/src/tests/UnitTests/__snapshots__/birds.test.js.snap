// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Renders <Login /> component correctly 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-justify-xs-center"
  >
    <div
      align="center"
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
    </div>
    <div
      align="center"
      className="MuiGrid-root MuiGrid-item"
    >
      <h1
        className="MuiTypography-root MuiTypography-h5"
      >
        Sign in
      </h1>
    </div>
  </div>
  <form
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
        htmlFor="outlined-name"
        id="outlined-name-label"
      >
        Username
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="username"
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          data-testid="username"
          disabled={false}
          id="outlined-name"
          name="username"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-7"
          >
            <span>
              Username
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
      >
        Password
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="current-password"
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          data-testid="password"
          disabled={false}
          name="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="password"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-7"
          >
            <span>
              Password
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            className="PrivateSwitchBase-input-12"
            data-indeterminate={false}
            onChange={[Function]}
            type="checkbox"
            value="remember"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Remember me
      </span>
    </label>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
      data-testid="submit"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Sign In
      </span>
    </button>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
          href="#"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Forgot password?
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
          href="/signup"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Don't have an account? Sign Up
        </a>
      </div>
    </div>
  </form>
</main>
`;
